lldb/source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -107,19 +107,18 @@ static ModuleSP findRuntime(Process &process, RuntimeKind runtime_kind) {
       return {};
   }
 
-  ModuleList images = process.GetTarget().GetImages();
-  for (unsigned i = 0, e = images.GetSize(); i < e; ++i) {
-    ModuleSP image = images.GetModuleAtIndex(i);
-    if (!image)
-      continue;
-    if (runtime_kind == RuntimeKind::Swift &&
-        IsModuleSwiftRuntime(process, *image))
-      return image;
-    if (runtime_kind == RuntimeKind::ObjC &&
-        objc_runtime->IsModuleObjCLibrary(image))
-      return image;
-  }
-  return {};
+  ModuleSP runtime_image;
+  process.GetTarget().GetImages().ForEach([&](const ModuleSP &image) {
+    if (image && ((runtime_kind == RuntimeKind::Swift &&
+                   IsModuleSwiftRuntime(process, *image)) ||
+                  (runtime_kind == RuntimeKind::ObjC && image &&
+                   objc_runtime->IsModuleObjCLibrary(image)))) {
+      runtime_image = image;
+      return false;
+    }
+    return true;
+  });
+  return runtime_image;
```

Comment:
are these changes NFC restructuring?

Reply:
Avoiding the ModuleList copy should be NFC.  The logic part also looks the same.  

The current code doesn't need to check `image` again in the RuntimeKind::ObjC clause.  

I'm not sure I'm in favor of this rewriting.  It does unify the return, but OTOT this is an order of magnitude harder to read than the old version IMO...

Reply:
Yes, see the commit message for the first of the two commits.

Reply:
```
Avoid copying the entire module list (NFC).  …
fa3181a
This does the exact same amount of locking as the previous version,
it's just slightly more efficient to use ForEach().
```

Reply:
These refactorings shall serve as proof that I actually read through the code ;-)

Reply:
I see now, thanks!

Reply:
Like Jim, I find the conditions in original form easier to read/understand.

Reply:
I'm not objecting to the ForEach use, to be clear.  Just combining the conditionals like that.